mlx/coverity.py

```diff
@@ -211,42 +211,45 @@ def process_coverity_nodes(app, doctree, fromdocname):
-                    row += cov_attribute_value_to_col(defect, 'Comment')
-                elif 'Classification' == item_col:
-                    row += cov_attribute_value_to_col(defect, 'Classification')
-                elif 'Action' == item_col:
-                    row += cov_attribute_value_to_col(defect, 'Action')
-                elif 'Status' == item_col:
-                    row += cov_attribute_value_to_col(defect, 'DefectStatus')
-                else:
-                    # generic check which if it is missing prints empty cell anyway
-                    row += cov_attribute_value_to_col(defect, item_col)
-
-            tbody += row
-        report_info(env, "done")
+        try:
+            for defect in defects['mergedDefects']:
+                row = nodes.row()
+
+                # go through each col and decide if it is there or we print empty
+                for item_col in node['col']:
+                    if 'CID' == item_col:
+                        # CID is default and even if it is in disregard
+                        row += create_cell(str(defect['cid']),
+                                           url=coverity_service.get_defect_url(app.config.coverity_credentials['stream'],  # noqa: E501
+                                                                               str(defect['cid'])))
+                    elif 'Category' == item_col:
+                        row += create_cell(defect['displayCategory'])
+                    elif 'Impact' == item_col:
+                        row += create_cell(defect['displayImpact'])
+                    elif 'Issue' == item_col:
+                        row += create_cell(defect['displayIssueKind'])
+                    elif 'Type' == item_col:
+                        row += create_cell(defect['displayType'])
+                    elif 'Checker' == item_col:
+                        row += create_cell(defect['checkerName'])
+                    elif 'Component' == item_col:
+                        row += create_cell(defect['componentName'])
+                    elif 'Comment' == item_col:
+                        row += cov_attribute_value_to_col(defect, 'Comment')
+                    elif 'Classification' == item_col:
+                        row += cov_attribute_value_to_col(defect, 'Classification')
+                    elif 'Action' == item_col:
+                        row += cov_attribute_value_to_col(defect, 'Action')
+                    elif 'Status' == item_col:
+                        row += cov_attribute_value_to_col(defect, 'DefectStatus')
+                    else:
+                        # generic check which if it is missing prints empty cell anyway
+                        row += cov_attribute_value_to_col(defect, item_col)
+
+                tbody += row
+            report_info(env, "done")
+        except AttributeError as e:
+            report_warning(env, 'Empty stream? Error: %s' % e)
         top_node += table
```

Comment:
No need to add an empty table when catching exception? Move this to the try-block as well?

Reply:
hm, well initial idea was to produce an empty table, which would indicate an empty stream. Would you assume that it might be better to put text describing that there was no data in stream?

Reply:
this is not necessarily a warning: an empty stream could also mean all coverity issues have been solved/triaged. So i think a message with 'no issues matching your query' or alike is  best

Reply:
Added. Might need a bit more pimping if we find this useful.